### ISSUE TITLE
Performance regression from Halide V14 -> V15 with compute_at()

### ISSUE THREAD

**Issue by catid** [halide/Halide#7623]
```
// Reproduces a regression in Halide V15:
// Using .compute_at() causes huge slow-downs.
class halide15_bug_generator : public Halide::Generator<halide15_bug_generator>
{
public:
    Input<Buffer<uint8_t>> input{"input", 2};
    Output<Buffer<uint8_t>> output{"output", 2};

    // Typically used for image coordinates
    Var x, y, xo, yo, xi, yi;

    Func intermediate{"intermediate"};

    void generate()
    {
        const int radius = 16;

        Func input_repeat = BoundaryConditions::repeat_edge(input);

        RDom r(-radius, radius*2);

        intermediate(x, y) = sum(2.3456f * cast<float>(input_repeat(x + r, y)));

        output(x, y) = saturating_cast<uint8_t>( intermediate(x, y) * 255.f );
    }

    void schedule()
    {
        const int vsize = natural_vector_size(output.type());

        output.tile(x, y, xo, yo, xi, yi, 64, 64)
            .vectorize(xi, vsize);

        // Adding this line causes V15 to go 5x slower.
        intermediate.compute_at(output, yi);
    }
};
```

**Comment by catid** [comment]
I provided a github repo that reproduces the problem on Ubuntu Linux here:

https://github.com/catid/halide-test/tree/bug15

This is the `bug15` branch of the repo.

**Comment by abadams** [comment]
Try adding the target flag enable_llvm_loop_opt

I believe it may have been on by default in 14 and off by default in 15, and it would have a large effect on that schedule, which seems to intentionally not vectorize the intermediate.

**Comment by catid** [comment]
That seems to fix it!  Why would you want to turn that off?

**Comment by catid** [comment]
Note that the auto-scheduler was also performing much worse without this flag too, so it seems to be important...

**Comment by abadams** [comment]
The flag adds a second layer of vectorization and unrolling, but controlled automatically by llvm. It doesn't do anything you can't do yourself in the schedule, and it sometimes does things you don't want that bloat code size for no gain, e.g. vectorizing loop tails, or unrolling loops that don't need to be unrolled.

It's a nice flag if you want semi-automated scheduling where you leave vectorization and unrolling decisions up to llvm.

**Comment by catid** [comment]
Maybe it's in a migration guide that I didn't find some some reason but wasted a bunch of time figuring this out and had to be told in an issue ticket.

**Comment by catid** [comment]
It would be really nice if this was called out in the release notes for the V15 series since it's probably going to be hit unknowingly by a bunch of other people.

**Comment by abadams** [comment]
This was a bad oversight. I've added a note for the 15.0.0 and 15.0.1 releases

**Comment by steven-johnson** [comment]
> This was a bad oversight. I've added a note for the 15.0.0 and 15.0.1 releases

Are we going to backport the fix and do a 15.0.2 release?

**Comment by abadams** [comment]
Assuming you're talking about the autoscheduler fix, it'll be fixed in the 16 release. Are there plausibly people who are interested in using a 15 point release after the 16 release is out?
